activerecord/lib/active_record/query_constraints.rb

```diff
@@ -39,7 +35,35 @@ module ClassMethods
       #   developer.reload
       #   # => SELECT "developers".* FROM "developers" WHERE "developers"."company_id" = 1 AND "developers"."id" = 1 LIMIT 1
       def query_constraints(*columns_list)
-        self.query_constraints_list = columns_list.map(&:to_s)
+        @_query_constraints_list = columns_list.map(&:to_s)
+      end
+
+      # Returns an array of strings with the names of the columns
+      # used to perform SELECT / UPDATE / DELETE queries on the model objects.
+      # Defaults to the primary key.
+      #
+      #   class Person < ActiveRecord::Base; end
+      #
+      #   Person.query_constraints_list # => ["id"]
+      #
+      #   class Comment < ActiveRecord::Base
+      #     query_constraints :post_id, :id
+      #   end
+      #
+      #   Comment.query_constraints_list # => ["post_id", "id"]
+      def query_constraints_list
```

Comment:
I don't think think should be part of the public API (at least not yet), so let's no-doc this. I don't really love "query_constraints_list" as an API and don't think applications should ever need to list out their query constraints list. It feels like this was implemented so we can test it, but I'd rather test that the queries do what we want rather than have Models output their query constraints.

Reply:
>It feels like this was implemented so we can test it

You are 100% right on this, I started from tests as I wanted to have a list of specs to make sure that inheritance works as we expect it and end up making the method public 🙃 
I'm going to change it to private and rewrite tests to assert for queries 👍